test/db/analysis/vars

```diff
@@ -873,10 +873,12 @@ pdf
 EOF
 EXPECT=<<EOF
             ;-- fn1:
-/ dbg.fn1 ();
-EXPECT=<<EOF
-            ;-- fn1:
-/ dbg.fn1 ();
+/ dbg.fn1 (int a, char *g, double q, struct Some *gg, some_t **out);
```

Comment:
Please add more tests based on issue https://github.com/rizinorg/rizin/issues/3540, just checking function signature and vars location should be enough:
```
[0x80003c60]> afv
arg fp_number_type * b @ a12
arg fp_number_type * a @ a13
arg fp_number_type * tmp @ a15
arg fractype a_fraction @ d11
arg int diff @ d12
arg int a_normal_exp @ d15
arg int b_normal_exp @ d2
arg int sdiff @ d3
arg fractype b_fraction @ d9
[0x80003c60]> afs
void dbg._fpadd_parts();
[0x80003c60]> pd 1
            ; CALL XREF from dbg.__adddf3 @ 0x80003e32
            ; CALL XREF from dbg.__subdf3 @ 0x80003e72
            ;-- _fpadd_parts:
╭ dbg._fpadd_parts (fp_number_type *a, fp_number_type *b, fp_number_type *tmp, int sdiff, int a_normal_exp, int b_normal_exp, fractype a_fraction, fractype b_fraction, int diff);
│ stack: 0 (vars 0, args 0)
│ rg: 9 (vars 0, args 9)
│           0x80003c60      0c40           ld.bu %d15, [%a4]0          ; fp-bit.c:604 ; fp_number_type * _fpadd_parts(fp_number_type * a,fp_number_type * b,fp_number_type * tmp);
[0x80003c60]>
```
Note, for some reason `afs` doesn't show the right signature, but that is a separate bug, should be done in another PR likely.

Reply:
Note also, that it's still different - it counted some vars as arguments, which is not true.

Reply:
Maybe because of this

```
RZ_API bool rz_analysis_var_is_arg(RzAnalysisVar *var) {
...
	switch (var->storage.type) {
	case RZ_ANALYSIS_VAR_STORAGE_REG:
		return true; // reg vars are always arguments for now
```

Reply:
cc @thestr4ng3r

Reply:
It should consult the calling convention in this function

Reply:
This should be helpful: https://www.infineon.com/dgdl/TriCore_EABI_v2_3.pdf